### PR TITLE
Windows integration test

### PR DESCRIPTION
**Pull request by andrewkroh**
Adding a simple integration test that writes messages to the Windows event log and then verifies that they can be read back. There are a lot more edge cases to test, but this is a start.

### CODE REVIEW

**Comment by andrewkroh** [comment]
@ruflin Any idea why I'm seeing 0% in the [report at codecov.io](https://codecov.io/github/elastic/winlogbeat?pr=8)? The reporting is happening from AppVeyor instead of Travis which is new for this PR. Also I am not (yet) doing any aggregation of the various coverage reports  generated during testing.

I archived the .cov file and .html file in the [AppVeyor build](https://ci.appveyor.com/project/elastic-beats/winlogbeat/build/artifacts) to help with troubleshooting.

**Comment by ruflin** [comment]
@andrewkroh The content of the integration.cov file looks good. So far I always used the bash script. I assume it has something to do with the pip script and the errors on lines 97, 111, 112. It seems like the params work differently here and the script always searchs for *.gcno files and probably ignores the -f param. Interestingly on line 115 is says it found the report ... @stevepeak can perhaps help here?

**Comment by andrewkroh** [comment]
I added the `-X gcov` flag to the codecov parameters and that resolved the errors from the command, but the [report](https://codecov.io/github/elastic/winlogbeat?pr=8) still shows 0% coverage.

**Comment by andrewkroh** [comment]
I marking this for review even though the code coverage is not fixed. It can be fixed in another PR. I want to get this merged so that I can move on.

**Comment by andrewkroh** [comment]
I found some problems and fixed them in the PR. I am working on a another PR now to cleanup all the code (docs, lint, etc) and log messages.

File: eventlog/eventlog_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+	log.Report(elog.Info, eventID, msgs)
+
+	// Read messages:
+	eventlog := NewEventLoggingAPI(providerName)
+	eventlog.Open(0)
+	defer eventlog.Close()
+	records, err := eventlog.Read()
+	if err != nil {
+		t.Error(err)
+		return
```
Comment (urso) [inline eventlog/eventlog_integration_test.go]: t.Error + return can be replaced by Fatal or Fatalf

**Comment by urso** [comment]
LGTM, but don't really know about the windows API.

File: eventlog/eventlog_integration_test.go (hunk 1 above)
Comment (andrewkroh) [inline eventlog/eventlog_integration_test.go]: @urso Good suggestion. I hope you don't mind if we merge this PR without this. I made this change in #9.

File: eventlog/eventlog_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+package eventlog
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"testing"
+
+	elog "github.com/andrewkroh/sys/windows/svc/eventlog"
```
Comment (ruflin) [inline eventlog/eventlog_integration_test.go]: should we move this to elastic?

**Comment by ruflin** [comment]
LGTM.

**Comment by stevepeak** [comment]
Hey @ruflin sorry I'm late to the party? :cake: 

Here is [the upload for c33f9915](https://codecov.s3.amazonaws.com/repos/4FB6777ED85D6FE64577BF7A47FB47A1/uploads/upload:310836:c33f9915527a39edf9b54eab86435afdc240cd78:1y6ots6p4n5lih20:677:1448311932.55.txt)

Looks like there is an issue with the infamous `\` backslashes. As we know Window and Linux are sadly still not friends.

Patching Codecov and will have the reports working for this asap.

Cheers!

**Comment by andrewkroh** [comment]
Thanks @stevepeak. The reports generated on Windows are working now.

/cc @ruflin
